homeassistant/components/mikrotik/__init__.py

```diff
@@ -1 +1,390 @@
+            vol.Any(MTK_LOGIN_PLAIN, MTK_LOGIN_TOKEN),
+        vol.Optional(CONF_PORT): cv.port,
+        vol.Optional(CONF_SSL, default=False): cv.boolean,
+        vol.Optional(CONF_ENCODING, default=DEFAULT_ENCODING): cv.string,
+        vol.Optional(CONF_TRACK_DEVICES, default=True): cv.boolean,
+        vol.Optional(CONF_ARP_PING, default=False): cv.boolean,
+        vol.Optional(CONF_SENSORS): vol.All(
+            cv.ensure_list, [vol.In(SENSORS)]),
+        vol.Optional(CONF_BINARY_SENSORS):
+            vol.All(cv.ensure_list, [vol.In(BINARY_SENSORS)]),
+        vol.Optional(CONF_WAN_PORT, default=MTK_DEFAULT_WAN): cv.string,
+    })
+)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.All(cv.ensure_list, [MIKROTIK_SCHEMA])
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Set up the Mikrotik component."""
+    hass.data[MIKROTIK] = {}
+    hass.data[CLIENT] = MikrotikAPI(hass, config[DOMAIN])
+
+    for device in config[DOMAIN]:
+        host = device[CONF_HOST]
+        hass.data[MIKROTIK][host] = {}
+        hass.data[CLIENT].connect_to_device(host)
+
+        _LOGGER.debug("[%s] Loading Mikrotik device intrgrations.", host)
+
+        sensors = device.get(CONF_SENSORS)
+        if sensors:
+            hass.async_create_task(
+                async_load_platform(
+                    hass, SENSOR, DOMAIN, device, config))
+
+        binary_sensors = device.get(CONF_BINARY_SENSORS)
+        if binary_sensors:
+            hass.async_create_task(
+                async_load_platform(
+                    hass, BINARY_SENSOR, DOMAIN, {
+                        CONF_BINARY_SENSORS: binary_sensors,
+                        CONF_HOST: host
+                    }, config))
+
+        if device[CONF_TRACK_DEVICES]:
+            hass.data[MIKROTIK][ARP] = {}
+            hass.data[MIKROTIK][DHCP] = {}
+            hass.async_create_task(
+                async_load_platform(
+                    hass, DEVICE_TRACKER, DOMAIN, device, config))
+
+    return True
+
+
+class MikrotikAPI:
+    """Handle all communication with the Mikrotik API."""
+
+    def __init__(self, hass, configs):
+        """Initialize the Mikrotik Client."""
+        self.hass = hass
+        self.config = {}
+        self._client = {}
+        self._hosts = {}
+        for config in configs:
+            host = config.get(CONF_HOST)
+            self._hosts[host] = {}
+            self._hosts[host]['config'] = config
+            self._hosts[host][CONNECTED] = False
+            self._hosts[host][CONNECTING] = False
+            self._hosts[host]['kwargs'] = None
+            self._hosts[host][CONF_WAN_PORT] = config.get(CONF_WAN_PORT)
+            self._hosts[host][DEVICE_TRACKER] = None
+            self._hosts[host][CONF_ARP_PING] = config.get(CONF_ARP_PING)
+
+    def config_kwargs(self, config):
+        """Build Mikrotik host config."""
+        from librouteros.login import login_plain, login_token
```

Comment:
3rd party imports can be moved to the top of the module.

Reply:
Moving to the top